compiler-plugin/src/test/kotlin/com/google/devtools/ksp/processor/AnnotatedUtilProcessor.kt

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.google.devtools.ksp.processor
+
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.google.devtools.ksp.symbol.KSNode
+import com.google.devtools.ksp.symbol.impl.getAnnotation
+import com.google.devtools.ksp.symbol.impl.getAnnotations
+import com.google.devtools.ksp.symbol.impl.getAnnotationsByType
+import com.google.devtools.ksp.symbol.impl.isAnnotationPresent
+import com.google.devtools.ksp.visitor.KSTopDownVisitor
+import kotlin.reflect.KClass
+
+class AnnotatedUtilProcessor : AbstractTestProcessor() {
+    val results = mutableListOf<String>()
+    private val annotationKClasses = listOf(ParametersTestAnnotation::class, ParameterArraysTestAnnotation::class, ParametersTestWithNegativeDefaultsAnnotation::class, OuterAnnotation::class)
+    private val visitors = listOf(GetAnnotationVisitor(), GetAnnotationsVisitor(), GetAnnotationsByTypeVisitor())
+
+
+    override fun process(resolver: Resolver): List<KSAnnotated> {
+        resolver.getSymbolsWithAnnotation("com.google.devtools.ksp.processor.Test", true).forEach {
+            results.add("Test: $it")
+            visitors.forEach { visitor -> it.accept(visitor, results) }
+        }
+        return emptyList()
+    }
+
+    override fun toResult(): List<String> {
+        return results
+    }
+
+    inner class GetAnnotationVisitor : KSTopDownVisitor<MutableCollection<String>, Unit>() {
+        override fun visitAnnotated(annotated: KSAnnotated, data: MutableCollection<String>) {
+            annotationKClasses.forEach { kClass ->
+                if (annotated.isAnnotationPresent(kClass)) {
+                    annotated.getAnnotation(kClass)?.let { data.add(it.asString()) }
+                }
+            }
+        }
+
+        override fun defaultHandler(node: KSNode, data: MutableCollection<String>) {
+        }
+    }
+
+    inner class GetAnnotationsVisitor : KSTopDownVisitor<MutableCollection<String>, Unit>() {
+        override fun visitAnnotated(annotated: KSAnnotated, data: MutableCollection<String>) {
+            annotated.getAnnotations().forEach { data.add(it.asString()) }
+        }
+
+        override fun defaultHandler(node: KSNode, data: MutableCollection<String>) {
+        }
+    }
+
+    inner class GetAnnotationsByTypeVisitor : KSTopDownVisitor<MutableCollection<String>, Unit>() {
+        override fun visitAnnotated(annotated: KSAnnotated, data: MutableCollection<String>) {
+            annotationKClasses.forEach { kClass ->
+                annotated.getAnnotationsByType(kClass).forEach { data.add(it.asString()) }
+            }
+        }
+
+        override fun defaultHandler(node: KSNode, data: MutableCollection<String>) {
+        }
+    }
+}
+
+@Suppress("LongParameterList")
+annotation class ParametersTestAnnotation(
```

Comment:
had to define the annotations twice (in test file and in processor) so the processor could use it in the getAnnotation and getAnnotationByType function calls